spock-core/src/main/java/org/spockframework/compiler/WhereBlockRewriter.java

```diff
@@ -381,6 +466,10 @@ private static void notAParameterization(ASTNode stat) throws InvalidSpecCompile
 "where-blocks may only contain parameterizations (e.g. 'salary << [1000, 5000, 9000]; salaryk = salary / 1000')");
   }
 
+  private static void dataTableHeaderMayOnlyContainVariableNames(ASTNode stat) throws InvalidSpecCompileException {
```

Comment:
might be better to return the exception so that you can directly see that it exits in the code

`throw dataTableHeaderMayOnlyContainVariableNames(column.get(0));`

Reply:
Sure, I totally agree, I just followed the tactic mandated for by `notAParameterization`.
I changed both now.